docs/source/setup.rst

```diff
@@ -0,0 +1,190 @@
+
+5. If you use you own program: Write you own ``runfile`` similar to ``nnlorun.py``.
+This script will be ran on each node, so it should be *self-contained* and
+*Python 2.4 compatible*. It should also be able to handle all arguments of
+the ``nnnnlorun.py``, even if they are not used in the script itself. It is
+easiest to simply copy the ``parse_arguments`` function from there. Most
+arguments correspond to a similar named setting from the runcard/header.
+To run on Dirac make sure you do not depend on a specific local setup, i.e.
+download all required programs from gfal or use what is available on the
+``/cvmfs/``.
+
+6. To install and run the scripts, run
+
+.. code-block:: bash
+
+    python3 setup.py install --user
+    python3 setup.py develop --user
+
+(Include the ``--prefix`` option and add to a location contained in ``$PYTHONPATH``
+if you want to install it elsewhere). ``--user`` is used on the gridui as we don't
+have access to the python3 installation - if you have your own install, feel free
+to drop it.
+We currently need to be in develop mode given the way that the header system works -
+the plan is for this to change at some point.
+
+Alternatively: if you wish to run pyHepGrid from within a Conda environment,
+install the scripts by moving to the directory containing setup.py and running:
+
+.. code-block:: bash
+
+    conda install conda-build
+    conda develop .
+
+If prompted to install any dependencies required by conda-build in step (1),
+type 'Y' to proceed.
+
+
+After this you should be able to run ``pyHepGrid test runcards/your_runcard.py
+-B``. This will execute the your ``runfile`` locally inside the ``test_sandbox``
+folder. You might want to try running it with a *clean* environment, e.g.
+without sourcing your ``~/.bashrc``. If this works fine you can try submitting to
+the arc  test queue with ``pyHepGrid run runcards/your_runcard.py -B --test``. The
+test queue highly limited in resources. **Only submit a few short jobs to it**
+(<10).
+
+Further customisations (advanced usage)
+=======================================
+
+Beside the header and runcard setup, ``pyHepGrid`` has two big *attack points* for
+customisations. First and foremost the ``runfile`` which is run on each grid node.
+This is similar to other grid-scripts that you might have used before. However
+you can also change some local background behaviour through ``runmodes``. A
+``runmode`` is *program* specific, e.g. there is a ``runmode`` ``"NNLOJET"`` and
+``"HEJ"``. The behaviour of ``pyHepGrid ini`` is completely controlled by a
+``runmode``. You could set it up to upload some common files (runcards,
+warmup-files, executable, etc.) with ``gfal`` before submitting jobs. An simple
+example for a completely customised ``runfile`` and ``runmode`` is provided in the
+``example`` folder.
+
+If you want to implement your own ``runmode`` write a *program* class as a
+subclass of the [`ProgramInterface`](../src/pyHepGrid/src/program_interface.py).
```

Comment:
What is the proper way of setting up these file links now?

Reply:
I have to find out!

Reply:
Found out it was too much work for three lines... just added the links manually.